state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -1816,17 +1942,37 @@ pub mod pallet {
 							},
 						};
 
-					// See if princial swap is needed, schedule it first if so:
+					// See if principal swap is needed, schedule it first if so:
 					if input_asset != output_asset && !principal_swap_amount.is_zero() {
+						let mut dca_state = DcaState {
+							scheduled_chunk_swap_id: None,
+							remaining_input_amount: principal_swap_amount,
+							// No DCA is equivalent to DCA with 1 chunk
+							remaining_chunks: dca_params.map(|p| p.number_of_chunks).unwrap_or(1),
```

Comment:
Just realised that the dca_params.chunk_interval isn't used anywhere... (and it's not mentioned in the PR description, I checked ;) )

Reply:
You are right, I missed this because the tests used the same value for chunk interval as the default SWAP_DELAY. Fixed now and updated the tests as well.